crypto/asn1/a_time.c

```diff
@@ -215,11 +216,11 @@ int asn1_time_to_tm(struct tm *tm, const ASN1_TIME *d)
         if (o + 4 != l)
             goto err;
         for (i = end; i < end + 2; i++) {
-            if ((a[o] < '0') || (a[o] > '9'))
+            if (!isdigit(a[o]))
                 goto err;
             n = a[o] - '0';
             o++;
-            if ((a[o] < '0') || (a[o] > '9'))
+            if (!isdigit(a[0]))
```

Comment:
Should this really be `a[0]` or should it be `a[o]` ?

If the latter, the test cases missed it.